vllm/model_executor/layers/fused_moe/fused_moe.py

```diff
@@ -129,7 +136,10 @@ def fused_moe_kernel(
                     mask=offs_k[:, None] < K - k * BLOCK_SIZE_K,
                     other=0.0)
         # We accumulate along the K dimension.
-        accumulator += tl.dot(a, b)
+        if use_fp8:
+            accumulator = tl.dot(a, b, acc=accumulator)
```

Comment:
Thanks for your great work! I have a question here, why does fp8 need to do this, is there any difference?

Reply:
This is a great question, I'm doing it because it supports the fp8 fast accumulation, see

https://github.com/NVIDIA/cutlass/blob/5c447dd84f8ae0e1d48ff9a2eae26ce8c4958101/CHANGELOG.md?plain=1#L67

https://github.com/openai/triton/blob/5623cdc5fb2d497d2d48cea89170707a97029219/python/triton/ops/matmul.py#L132

The downside is that you can't do rescaling before adding it to the accumulator